pkg/cluster/spec/tiflash.go

```diff
@@ -241,19 +242,26 @@ server_configs:
 }
 
 // InitTiFlashLearnerConfig initializes TiFlash learner config file
-func (i *TiFlashInstance) InitTiFlashLearnerConfig(cfg *scripts.TiFlashScript, src map[string]interface{}) (map[string]interface{}, error) {
+func (i *TiFlashInstance) InitTiFlashLearnerConfig(cfg *scripts.TiFlashScript, clusterVersion string, src map[string]interface{}) (map[string]interface{}, error) {
 	topo := Specification{}
+	var statusAddr string
 
 	firstDataDir := strings.Split(cfg.DataDir, ",")[0]
 
+	if semver.Compare("v4.0.5", clusterVersion) <= 0 {
```

Comment:
Prefer 
```suggestion
	if semver.Compare(clusterVersion, "v4.0.5") >= 0 {
```

Reply:
> An invalid semantic version string is considered less than a valid one.

And "nightly" is an invalid one, do we need to handle it specifically?

Reply:
nightly version is not affected. This config change is usable since v4.0.4, so is nightly.